guides/search.md

```diff
@@ -0,0 +1,381 @@
+```
+
+With sorting, you can also use the `search_after` parameter to paginate through the search results. Let's say you have already displayed the first page of results, and you want to display the next page. You can use the `search_after` parameter to paginate through the search results. The following example will demonstrate how to get the first 3 pages of results using the search query of the previous example:
+
+```rust
+let page_1 = client
+    .search(SearchParts::Index(&["movies"]))
+    .from(5)
+    .size(2)
+    .body(json!({
+        "query": {
+            "match": {
+                "title": "dark knight"
+            }
+        },
+        "sort": [
+            {
+                "year": {
+                    "order": "asc"
+                }
+            }
+        ]
+    }))
+    .send()
+    .await?
+    .json::<Value>()
+    .await?;
+let page_1_hits = page_1["hits"]["hits"].as_array().unwrap();
+println!("{:?}", page_1_hits);
+
+let page_2 = client
+    .search(SearchParts::Index(&["movies"]))
+    .size(2)
+    .body(json!({
+        "query": {
+            "match": {
+                "title": "dark knight"
+            }
+        },
+        "sort": [
+            {
+                "year": {
+                    "order": "asc"
+                }
+            }
+        ],
+        "search_after": page_1_hits[page_1_hits.len() - 1]["sort"]
+    }))
+    .send()
+    .await?
+    .json::<Value>()
+    .await?;
+let page_2_hits = page_2["hits"]["hits"].as_array().unwrap();
+println!("{:?}", page_2_hits);
+
+let page_3 = client
+    .search(SearchParts::Index(&["movies"]))
+    .size(2)
+    .body(json!({
+        "query": {
+            "match": {
+                "title": "dark knight"
+            }
+        },
+        "sort": [
+            {
+                "year": {
+                    "order": "asc"
+                }
+            }
+        ],
+        "search_after": page_2_hits[page_2_hits.len() - 1]["sort"]
+    }))
+    .send()
+    .await?
+    .json::<Value>()
+    .await?;
+let page_3_hits = page_3["hits"]["hits"].as_array().unwrap();
+println!("{:?}", page_3_hits);
+```
+
+### Pagination with scroll
+
+When retrieving large amounts of non-real-time data, you can use the `scroll` parameter to paginate through the search results.
+
+```rust
+let page_1 = client
+    .search(SearchParts::Index(&["movies"]))
+    .scroll("1m")
+    .size(2)
+    .body(search_body)
+    .send()
+    .await?
+    .json::<Value>()
+    .await?;
+
+let page_2 = client
+    .scroll(ScrollParts::None)
+    .body(json!({
+        "scroll_id": page_1["_scroll_id"].as_str().unwrap(),
+        "scroll": "1m"
+    }))
+    .send()
+    .await?
+    .json::<Value>()
+    .await?;
+
+let page_3 = client
+    .scroll(ScrollParts::None)
+    .body(json!({
+        "scroll_id": page_2["_scroll_id"].as_str().unwrap(),
+        "scroll": "1m"
+    }))
+    .send()
+    .await?
+    .json::<Value>()
+    .await?;
```

Comment:
Please inline the search body, additionally `scroll_id` and `scroll` on the `scroll` operation are query parameters set via functions, not in the body:
```suggestion
let page_1 = client
    .search(SearchParts::Index(&["movies"]))
    .scroll("1m")
    .size(2)
    .body(json!({
        "query": {
            "match": {
                "title": "dark knight"
            }
        },
        "sort": [
            {
                "year": {
                    "order": "asc"
                }
            }
        ]
    }))
    .send()
    .await?
    .json::<Value>()
    .await?;

let page_2 = client
    .scroll(ScrollParts::None)
    .scroll_id(page_1["_scroll_id"].as_str().unwrap())
    .scroll("1m")
    .send()
    .await?
    .json::<Value>()
    .await?;

let page_3 = client
    .scroll(ScrollParts::None)
    .scroll_id(page_2["_scroll_id"].as_str().unwrap())
    .scroll("1m")
    .send()
    .await?
    .json::<Value>()
    .await?;
```